ts/components/FiscalCodeLandscapeOverlay.tsx

```diff
@@ -52,93 +53,132 @@ const styles = StyleSheet.create({
-    this.props.onCancel();
-    return true;
-  };
+const FiscalCodeLandscapeOverlay: React.FunctionComponent<Props> = (
+  props: Props
+) => {
+  // eslint-disable-next-line functional/no-let
+  let scrollTimeout: number | undefined;
 
-  public componentDidMount() {
-    BackHandler.addEventListener("hardwareBackPress", this.handleBackPress);
-  }
+  const ScrollViewRef = React.createRef<ScrollView>();
 
-  public componentWillUnmount() {
-    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
-    // if there is an active timeout, clear it!
-    if (this.scrollTimeout !== undefined) {
-      clearTimeout(this.scrollTimeout);
-      // eslint-disable-next-line
-      this.scrollTimeout = undefined;
-    }
-  }
+  const handleBackPress = () => {
+    // On backpress the component gets unmounted, so the brightness is restored by the
+    // cleanup function
+    props.onCancel();
+    return true;
+  };
 
-  private scrollToEnd = () => {
-    if (this.props.showBackSide && this.ScrollVewRef.current) {
+  const scrollToEnd = () => {
+    if (props.showBackSide && ScrollViewRef.current) {
       // dalay the scroll to end command to wait until the ingress animation is completed
       // eslint-disable-next-line
-      this.scrollTimeout = setTimeout(() => {
-        if (this.ScrollVewRef.current) {
-          this.ScrollVewRef.current.scrollToEnd({ animated: true });
+      scrollTimeout = setTimeout(() => {
+        if (ScrollViewRef.current) {
+          ScrollViewRef.current.scrollToEnd({ animated: true });
         }
       }, 300);
     }
   };
 
-  public render() {
-    return (
-      <Container style={{ backgroundColor: customVariables.brandDarkGray }}>
-        <AppHeader noLeft={true} dark={true}>
-          <Body />
-        </AppHeader>
-        <StatusBar
-          backgroundColor={customVariables.brandDarkGray}
-          barStyle={"light-content"}
-        />
-        <ScrollView
-          style={styles.content}
-          ref={this.ScrollVewRef}
-          onLayout={this.scrollToEnd}
-        >
-          <View style={styles.headerSpacer} />
-          <View spacer={true} />
-          <View>
-            <FiscalCodeComponent
-              type={"Landscape"}
-              profile={this.props.profile}
-              getBackSide={false}
-              municipality={this.props.municipality}
-            />
-          </View>
-
-          <View spacer={true} />
+  React.useEffect(() => {
+    BackHandler.addEventListener("hardwareBackPress", handleBackPress);
+    return () => {
+      BackHandler.removeEventListener("hardwareBackPress", handleBackPress);
+      // if there is an active timeout, clear it!
+      if (scrollTimeout !== undefined) {
+        clearTimeout(scrollTimeout);
+        // eslint-disable-next-line
+        scrollTimeout = undefined;
+      }
+    };
+  }, []);
+
+  // Brightness effect manager
+  React.useEffect(() => {
+    const myBrightF = async () =>
```

Comment:
In this way if the `getBrightness` fails we force the device to have a brightness of 0.5
This could be not true

Perhaps we could return undefined in left case?

```typescript
const myBrightF = async () =>
      await getBrightness()
        .fold(
          () => undefined, // If fails, we don't know the brightness value
          _ => _
        )
        .run();


[...]

return () => {
      const restoreDeviceBrightnessF = async () => {
        const deviceB = await myBright;
        if (deviceB) {
          await setBrightness(deviceB)
            .fold(
              () => alert("Can't restore brightness"),
              _ => _
            )
            .run();
        }
      };
      void restoreDeviceBrightnessF();
    };
```

Reply:
Thank you, this is a good idea. I'll refactor the code as you suggest. I set it to a default because that value is used to construct the cleanup function, so I thought it couldn't be undefined